worker/lib/workers/qemu.ts

```diff
@@ -76,6 +80,19 @@ class QemuWorker extends EventEmitter implements Leviathan.Worker {
 		if (this.screenCapturer != null) {
 			await this.screenCapturer.teardown();
 		}
+
+		// remove iptables rules set up from host
+		try{
+			await execProm(`iptables-legacy-save | grep -v 'comment ${this.iptablesComment}' | iptables-legacy-restore`)
```

Comment:
Need to test this command (without the grep) on real jenkins-agent shell.

https://dashboard.balena-cloud.com/devices/11a1d0258e8782b188c3598b57c155c6/summary

Reply:
```
jenkins@f8d197dc699e:~$ iptables-legacy-save | iptables-legacy-restore
Failed to list table names in /proc/net/ip_tables_names: Permission denied
```

Reply:
Seems to work as root, just not as the jenkins user. This would have been a similar issue on your workstation so we must be running as root somewhere?
```
sudo iptables-legacy-save | sudo iptables-legacy-restore
```

Reply:
Can we also try to run this command on teardown? Ideally we would clean up rules on start and stop of the worker.

Reply:
@klutchell It works from within the worker container locally because we have this mapping in the docker-compose I believe:

```
"/dev/net/tun:/dev/net/tun"
```

But if the jenkins agent shell that is managing the container doesn't have the right permissions, then maybe that will still stop it working, I'll have a look

Reply:
If we are running `docker-compose up` from the jenkins agent and have permissions to do that, then I guess it doesn't matter what other permissions are available since it's now up to our compose file?

It looks like we are using docker-in-docker on the agent so we really just need to test it I think.
https://github.com/product-os/balena-jenkins/blob/master/docker-compose.yml#L192

Reply:
> It works from within the worker container locally because we have this mapping in the docker-compose I believe

The tun mapping is for QEMU TUN/TAP networking, it works locally because the container is using host mode networking with the NET_ADMIN capability